### DISCUSSION TITLE
Window load event listener not running from rust.

### DISCUSSION THREAD

**Discussion by vjackson725** [wasm-bindgen/wasm-bindgen#4375] [Q&A] [answered]
Hi all,

I've been trying to run a closure on window load, but it's just not running for me, and I'm not sure why.

Here's the code which attaches the listener
```
  let callback = {
    Closure::<dyn Fn()>::new(move || {
      console::log_1(&"loaded (from Rust)".into());
    })
  };
  let _ = {
    window()
      .expect("window")
      .add_event_listener_with_callback(
        "load",
        JsCast::unchecked_ref(callback.as_ref())
      )
  };
  callback.forget();
```
(full code here: https://github.com/vjackson725/wasm_tests/tree/main/onload)

I also attached an event listener directly using javascript, and that worked, so I'm not sure what's going wrong here. Am I doing something wrong, or is something else going on?

**Comment by daxpedda** [comment, marked as answer]
That's because the `Window` has already loaded at the point where Wasm is executed.
You can check interactively if the `Window` is loaded via [`Document.readyState`](https://developer.mozilla.org/en-US/docs/Web/API/Document/readyState).

**Reply by vjackson725** [reply]
Ah, that makes perfect sense. Thanks for the help!
